kopf/reactor/peering.py

```diff
@@ -100,34 +102,46 @@ def keepalive(self):
         Add a peer to the peers, and update its alive status.
         """
         self.touch()
-        apply_peers([self], peering=self.peering)
+        apply_peers([self], peering=self.peering, namespace=self.namespace)
 
     def disappear(self):
         """
         Remove a peer from the peers (gracefully).
         """
         self.touch(lifetime=0)
-        apply_peers([self], peering=self.peering)
+        apply_peers([self], peering=self.peering, namespace=self.namespace)
 
 
 def apply_peers(
         peers: Iterable[Peer],
         peering: str,
+        namespace: Union[None, str],
 ):
     """
     Apply the changes in the peers to the sync-object.
 
     The dead peers are removed, the new or alive peers are stored.
     Note: this does NOT change their `lastseen` field, so do it explicitly with ``touch()``.
     """
+    body = {'status': {peer.id: None if peer.is_dead else peer.as_dict() for peer in peers}}
     api = kubernetes.client.CustomObjectsApi()
-    api.patch_cluster_custom_object(
-        group=PEERING_CRD_RESOURCE.group,
-        version=PEERING_CRD_RESOURCE.version,
-        plural=PEERING_CRD_RESOURCE.plural,
-        name=peering,
-        body={'status': {peer.id: None if peer.is_dead else peer.as_dict() for peer in peers}},
-    )
+    if namespace is None:
```

Comment:
what about wrapping this logic in a class? so we can use it later to fetch/patch peering custom objects.
 e.g.
```
PeeringObjects.patch(body, peering, namespace: Optional[String]) 
PeeringObjects.exists(peering, namespace: Optional[String]) -> bool
```

Reply:
[`pykube-ng`](https://github.com/hjacobs/pykube) does that, more or less. I prefer to not re-invent it or any other Kubernetes client. Instead, it is better to built on top of pykube-ng (see #15). I have a draft branch, but there are few issues both in pykube-ng, and in pykube-to-kopf connectivity — I am going to get back to them later, after I cover the functionality with the tests.